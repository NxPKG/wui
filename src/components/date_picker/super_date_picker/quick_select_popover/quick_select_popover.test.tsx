/*
 * Copyright 2022 Wazuh Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * NOTICE: THIS FILE HAS BEEN MODIFIED BY WAZUH INC UNDER COMPLIANCE WITH THE APACHE 2.0 LICENSE FROM THE ORIGINAL WORK
 * OF THE COMPANY Elasticsearch B.V.
 *
 * THE FOLLOWING IS THE COPYRIGHT OF THE ORIGINAL DOCUMENT:
 *
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';

import {
  WuiQuickSelectPopover,
  WuiQuickSelectPopoverProps,
} from './quick_select_popover';

const noop = () => {};

const defaultProps: WuiQuickSelectPopoverProps = {
  applyTime: noop,
  applyRefreshInterval: noop,
  start: 'now-15m',
  end: 'now',
  isDisabled: false,
  isPaused: true,
  refreshInterval: 0,
  commonlyUsedRanges: [{ start: 'now/d', end: 'now/d', label: 'Today' }],
  dateFormat: 'MMM D, YYYY @ HH:mm:ss.SSS',
  recentlyUsedRanges: [{ start: 'now/d', end: 'now/d', label: 'Today' }],
  isAutoRefreshOnly: false,
};

describe('WuiQuickSelectPopover', () => {
  test('is rendered', () => {
    const component = shallow(<WuiQuickSelectPopover {...defaultProps} />);

    expect(component).toMatchSnapshot();
  });

  test('isAutoRefreshOnly', () => {
    const component = shallow(
      <WuiQuickSelectPopover {...defaultProps} isAutoRefreshOnly={true} />
    );

    expect(component).toMatchSnapshot();
  });
});
