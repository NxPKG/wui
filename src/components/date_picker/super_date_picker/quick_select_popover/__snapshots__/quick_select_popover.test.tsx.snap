// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WuiQuickSelectPopover is rendered 1`] = `
<WuiPopover
  anchorClassName="wuiQuickSelectPopover__anchor"
  anchorPosition="downLeft"
  button={
    <WuiButtonEmpty
      aria-label="Date quick select"
      className="wuiFormControlLayout__prepend"
      data-test-subj="superDatePickerToggleQuickMenuButton"
      iconSide="right"
      iconType="arrowDown"
      isDisabled={false}
      onClick={[Function]}
      size="xs"
      textProps={
        Object {
          "className": "wuiQuickSelectPopover__buttonText",
        }
      }
    >
      <WuiIcon
        type="calendar"
      />
    </WuiButtonEmpty>
  }
  closePopover={[Function]}
  display="inlineBlock"
  hasArrow={true}
  id="QuickSelectPopover"
  isOpen={false}
  ownFocus={true}
  panelPaddingSize="m"
>
  <div
    className="wuiQuickSelectPopover__content"
    data-test-subj="superDatePickerQuickMenu"
  >
    <WuiQuickSelect
      applyTime={[Function]}
      end="now"
      start="now-15m"
    />
    <WuiCommonlyUsedTimeRanges
      applyTime={[Function]}
      commonlyUsedRanges={
        Array [
          Object {
            "end": "now/d",
            "label": "Today",
            "start": "now/d",
          },
        ]
      }
    />
    <WuiRecentlyUsed
      applyTime={[Function]}
      commonlyUsedRanges={
        Array [
          Object {
            "end": "now/d",
            "label": "Today",
            "start": "now/d",
          },
        ]
      }
      dateFormat="MMM D, YYYY @ HH:mm:ss.SSS"
      recentlyUsedRanges={
        Array [
          Object {
            "end": "now/d",
            "label": "Today",
            "start": "now/d",
          },
        ]
      }
    />
    <WuiRefreshInterval
      applyRefreshInterval={[Function]}
      isPaused={true}
      refreshInterval={0}
    />
  </div>
</WuiPopover>
`;

exports[`WuiQuickSelectPopover isAutoRefreshOnly 1`] = `
<WuiPopover
  anchorClassName="wuiQuickSelectPopover__anchor"
  anchorPosition="downLeft"
  button={
    <WuiButtonEmpty
      aria-label="Date quick select"
      className="wuiFormControlLayout__prepend"
      data-test-subj="superDatePickerToggleQuickMenuButton"
      iconSide="right"
      iconType="arrowDown"
      isDisabled={false}
      onClick={[Function]}
      size="xs"
      textProps={
        Object {
          "className": "wuiQuickSelectPopover__buttonText",
        }
      }
    >
      <WuiIcon
        type="clock"
      />
    </WuiButtonEmpty>
  }
  closePopover={[Function]}
  display="inlineBlock"
  hasArrow={true}
  id="QuickSelectPopover"
  isOpen={false}
  ownFocus={true}
  panelPaddingSize="m"
>
  <div
    className="wuiQuickSelectPopover__content"
    data-test-subj="superDatePickerQuickMenu"
  >
    <WuiRefreshInterval
      applyRefreshInterval={[Function]}
      isPaused={true}
      refreshInterval={0}
    />
  </div>
</WuiPopover>
`;
